var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -123,6 +124,15 @@ class Hpx(CMakePackage, CudaPackage, ROCmPackage):
         conflicts("cxxstd=14")
         depends_on("cuda@11:", when="+cuda")
 
+    # Restrictions for 1.8.X
```

Comment:
I can't make a change suggestion on the lines outside of the changed part, but this should look roughly like this: https://github.com/spack/spack/commit/925e5b17fc6a4e1a6d51a2651c7ca0ab0bec1a99. Basically, the asio, boost, and cmake version requirements are the same as 1.7.0 and don't need to be repeated. According to the release notes clang 8 and older is no longer supported.

Reply:
Likewise `depends_on("cuda@11:", when="+cuda")` needs to be in the 1.8 section.